subprojects/build-init/src/main/groovy/org/gradle/buildinit/plugins/internal/GroovyLibraryProjectInitDescriptor.java

```diff
@@ -16,43 +16,22 @@
 
-    public GroovyLibraryProjectInitDescriptor(TemplateOperationFactory templateOperationFactory, FileResolver fileResolver,
-                                              TemplateLibraryVersionProvider libraryVersionProvider, ProjectInitDescriptor globalSettingsDescriptor, DocumentationRegistry documentationRegistry) {
-        super("groovy", templateOperationFactory, fileResolver, libraryVersionProvider, globalSettingsDescriptor);
-        this.documentationRegistry = documentationRegistry;
+    public GroovyLibraryProjectInitDescriptor(TemplateOperationFactory templateOperationFactory, FileResolver fileResolver, TemplateLibraryVersionProvider libraryVersionProvider, ProjectInitDescriptor projectInitDescriptor, DocumentationRegistry documentationRegistry) {
+        super(templateOperationFactory, fileResolver, libraryVersionProvider, projectInitDescriptor, documentationRegistry);
     }
 
     @Override
-    public void generate(BuildInitTestFramework testFramework) {
-        globalSettingsDescriptor.generate(testFramework);
-
-        BuildScriptBuilder buildScriptBuilder = new BuildScriptBuilder(fileResolver.resolve("build.gradle"))
-            .fileComment("This generated file contains a sample Groovy project to get you started.")
-            .fileComment("For more details take a look at the Groovy Quickstart chapter in the Gradle")
-            .fileComment("user guide available at " + documentationRegistry.getDocumentationFor("tutorial_groovy_projects"))
-            .plugin("Apply the groovy plugin to add support for Groovy", "groovy")
-            .compileDependency("Use the latest Groovy version for building this library",
-                "org.codehaus.groovy:groovy-all:" + libraryVersionProvider.getVersion("groovy"))
-            .testCompileDependency("Use the awesome Spock testing and specification framework",
-                "org.spockframework:spock-core:" + libraryVersionProvider.getVersion("spock"),
-                "junit:junit:" + libraryVersionProvider.getVersion("junit"));
-        buildScriptBuilder.create().generate();
-
-        TemplateOperation groovyLibTemplateOperation = fromClazzTemplate("groovylibrary/Library.groovy.template", "main");
-        TemplateOperation groovyTestTemplateOperation = fromClazzTemplate("groovylibrary/LibraryTest.groovy.template", "test");
-        whenNoSourcesAvailable(groovyLibTemplateOperation, groovyTestTemplateOperation).generate();
+    protected TemplateOperation sourceTemplateOperation() {
+        return fromClazzTemplate("groovylibrary/Library.groovy.template", "main");
     }
 
     @Override
-    public boolean supports(BuildInitTestFramework testFramework) {
-        return testFramework == BuildInitTestFramework.SPOCK;
+    protected TemplateOperation testTemplateOperation(BuildInitTestFramework testFramework) {
+        return fromClazzTemplate("groovylibrary/LibraryTest.groovy.template", "test");
```

Comment:
The other comment in now in the outdated section, but the check for the testing framework already occurs in `InitBuild.java`.